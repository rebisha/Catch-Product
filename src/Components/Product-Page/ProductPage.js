import React, { useEffect, useState } from "react";
// components
import ProductCollection from "../Product-Collection/ProductCollection";
import Select from "../Select/Select";

import styles from "./productPage.scss";

const ProductPage = () => {
  const [product, setProduct] = useState([]);
  const [sortProduct, setSortProduct] = useState();

  useEffect(() => {
    const fetchData = async () => {
      try {
        const fetchProduct = await fetch(
          "http://catch-code-challenge.s3-website-ap-southeast-2.amazonaws.com/challenge-3/response.json"
        );
        const data = await fetchProduct.json();
        setProduct(data.results);
      } catch (error) {
        console.log(error);
      }
    };
    fetchData();
  }, []);

  const handleChange = e => {
    setSortProduct(e.target.value);
    switch (e.target.value) {
      case "lowest":
        return setProduct(product.sort((a, b) => a.salePrice - b.salePrice));
      case "highest":
        return setProduct(product.sort((a, b) => b.salePrice - a.salePrice));
      default:
        return product;
    }
  };

  return (
    <section className={styles.productPage}>
      <Select onChange={handleChange} />
      <ProductCollection products={product} />
    </section>
  );
};

export default ProductPage;
