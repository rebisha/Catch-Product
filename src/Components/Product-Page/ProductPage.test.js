import React from "react";
import { fetch } from "whatwg-fetch";
import renderer from "react-test-renderer";
// component
import ProductPage from "./ProductPage";

it("Calls api to get list of products", async () => {
  expect.assertions(2);

  const fetchProduct = await fetch(
    "http://catch-code-challenge.s3-website-ap-southeast-2.amazonaws.com/challenge-3/response.json"
  );
  const data = await fetchProduct.json();
  const products = data.results;

  expect(products).toBe(products);
  expect(products.length).toEqual(36);
});

it("tests error with async/await", async () => {
  expect.assertions(2);
  try {
    const fetchProduct = await fetch(
      "http://catch-code-challenge.s3-website-ap-southeast-2.amazonaws.com/challenge-3/response.json"
    );
    const data = await fetchProduct.json();
    const products = data.results;
    expect(products).toBe(false);
  } catch (error) {
    expect(error).toBe(error);
  }
});

it("Should render correctly", () => {
  const element = renderer.create(<ProductPage />);
  expect(element).toMatchSnapshot();
});
