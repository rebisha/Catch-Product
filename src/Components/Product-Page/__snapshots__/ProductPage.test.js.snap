// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly 1`] = `
<section
  className="productPage"
>
  <div
    className="select"
  >
    <select
      className="select__select"
      id="sorting"
      name="sorting"
      onChange={[Function]}
    >
      <option
        value=""
      >
        Sort by price
      </option>
      <option
        value="highest"
      >
        Highest Price
      </option>
      <option
        value="lowest"
      >
        Lowest Price
      </option>
    </select>
  </div>
  <section
    className="productCollection"
  />
</section>
`;
